fido2/tpm.py

```diff
@@ -0,0 +1,455 @@
+        if tpmi_st_attest != cls.TPM_ST_ATTEST_CERTIFY:
+            raise ValueError("tpmi_st_attest field is invalid")
+
+        try:
+            name = reader.read(reader.unpack("!H"))
+            data = reader.read(reader.unpack("!H"))
+
+            clock = reader.unpack("!Q")
+            reset_count = reader.unpack("!L")
+            restart_count = reader.unpack("!L")
+            safe_value = reader.unpack("B")
+            if safe_value not in (0, 1):
+                raise ValueError("invalid value 0x{0:x} for boolean".format(safe_value))
+            safe = safe_value == 1
+
+            firmware_version = reader.unpack("!Q")
+
+            attested_name = reader.read(reader.unpack("!H"))
+            attested_qualified_name = reader.read(reader.unpack("!H"))
+        except struct.error as e:
+            raise ValueError(e)
+
+        return cls(
+            name=name,
+            data=data,
+            clock_info=(clock, reset_count, restart_count, safe),
+            firmware_version=firmware_version,
+            attested=(attested_name, attested_qualified_name),
+        )
+
+    def __init__(self, name, data, clock_info, firmware_version, attested):
+        self.name = name
+        self.data = data
+        self.clock_info = clock_info
+        self.firmware_version = firmware_version
+        self.attested = attested
+
+    def __repr__(self):
+        return (
+            "<TpmAttestationFormat"
+            " data={self.data}"
+            " name={self.name}"
+            " clock_info={self.clock_info}"
+            " firmware_version=0x{self.firmware_version:x}"
+            " attested={self.attested}"
+            ">".format(self=self)
+        )
+
+
+class TpmsRsaParms(object):
+    """ Parse TPMS_RSA_PARMS struct
+
+    See:
+    https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
+    section 12.2.3.5
+    """
+
+    @classmethod
+    def parse(cls, reader, attributes):
+        ATTRIBUTES = TpmPublicFormat.ATTRIBUTES
+
+        symmetric = reader.unpack("!H")
+
+        restricted_decryption = attributes & (
+            ATTRIBUTES.RESTRICTED | ATTRIBUTES.DECRYPT
+        )
+        is_restricted_decryption_key = restricted_decryption == (
+            ATTRIBUTES.DECRYPT | ATTRIBUTES.RESTRICTED
+        )
+        if not is_restricted_decryption_key and symmetric != TPM_ALG_NULL:
+            # if the key is not a restricted decryption key, this field
+            # shall be set to TPM_ALG_NULL.
+            raise ValueError("symmetric is expected to be NULL")
+        # Otherwise should be set to a supported symmetric algorithm, keysize and mode
+        # TODO(baloo): Should we have non-null value here, do we expect more data?
+
+        scheme = reader.unpack("!H")
+
+        restricted_sign = attributes & (ATTRIBUTES.RESTRICTED | ATTRIBUTES.SIGN_ENCRYPT)
+        is_unrestricted_signing_key = restricted_sign == ATTRIBUTES.SIGN_ENCRYPT
```

Comment:
the kind of use-case I don't know how to achieve with `IntEnum`

Reply:
That should actually work fine as-is with IntEnum. The only difference being that `restricted_sign` and `attributes` won't be instances of ATTRIBUTES, but of int. The bit-wise operations and the equality check should still function the same.